src/components/helpers/wrapWithApiConnect.js

```diff
@@ -1,81 +1,62 @@
-import { Component, createElement } from 'react'
-
-import invariant from 'invariant'
 import hoistStatics from 'hoist-non-react-statics'
-import { forEach } from 'lodash'
+import React, { useEffect, useRef } from 'react'
 
 import { promisePropsEqual } from '../../utils'
-import getDisplayName from './getDisplayName'
 import { ELEMENT_ID_PROP_NAME } from './constants'
+import getDisplayName from './getDisplayName'
 
 const wrapWithApiConnect = ({
   finalMapPropsToPromiseProps,
-  withRef,
 }) => WrappedComponent => {
-  class ApiConnect extends Component {
-    componentWillMount() {
-      this.loadEntities(this.props)
-    }
+  function ApiConnect({ [ELEMENT_ID_PROP_NAME]: elementId, innerRef, ...rest }) {
```

Comment:
not sure why the linter doesn't catch it, but this looks like an unused variable to me. For omitting props, I like to use the convention of mapping to a name starting with an underscore, `_`, `_1`, ... eslint would understand this:

```suggestion
  function ApiConnect({ [ELEMENT_ID_PROP_NAME]: _, innerRef, ...rest }) {
```

Reply:
Thanks for the suggestion. I've implemented your change.